File: python/leetcode/question225.py
'''
请你仅使用两个队列实现一个后入先出（LIFO）的栈，并支持普通栈的全部四种操作（push、top、pop 和 empty）。

实现 MyStack 类：

void push(int x) 将元素 x 压入栈顶。
int pop() 移除并返回栈顶元素。
int top() 返回栈顶元素。
boolean empty() 如果栈是空的，返回 true ；否则，返回 false 。
'''
class MyStack:

    def __init__(self):
        self.list = []
        

    def push(self, x: int) -> None:
        self.list.append(x)
        

    def pop(self) -> int:
        if self.empty() :
            raise  Exception("数组为空")
        result = self.list.pop()
        return result

    def top(self) -> int:
        if self.empty() :
            raise  Exception("数组为空")
        return self.list[-1]

    def empty(self) -> bool:
        return len(self.list) == 0
        


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: fronted/typescript/leetcode/question225.ts
class MyStack {
    private list: Array<number>;

    constructor() {
        this.list = []
    }

    push(x: number): void {
        this.list.push(x)
    }

    pop(): number {
        if (this.list.length === 0) {
            throw new Error("数组不存在")
        }
        return this.list.pop() as number
    }

    top(): number {
        if (this.list.length === 0) {
            throw new Error("数组不存在")
        }
        return this.list[this.list.length - 1]
    }

    empty(): boolean {
        return this.list.length === 0
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * var obj = new MyStack()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */